cineast-core/src/main/java/org/vitrivr/cineast/core/features/abstracts/AbstractSegmentExporter.java

```diff
@@ -0,0 +1,124 @@
+
+  protected static final Logger LOGGER = LogManager.getLogger();
+
+  private static final String PROPERTY_NAME_DESTINATION = "destination";
+
+  /**
+   * Destination path for the audio-segment.
+   */
+  private final Path destination;
+
+  protected String fileExtension;
+
+  protected String dataUrlPrefix;
+
+  /**
+   * Default constructor
+   */
+  public AbstractSegmentExporter() {
+    this(new HashMap<>());
+  }
+
+  /**
+   * Default constructor. The AudioSegmentExport can be configured via named properties in the provided HashMap.
+   * <p>
+   * Supported parameters:
+   *
+   * <ol>
+   *      <li>destination: Path where files should be stored.</li>
+   * </ol>
+   *
+   * @param properties HashMap containing named properties
+   */
+  public AbstractSegmentExporter(HashMap<String, String> properties) {
+    this.destination = Path.of(properties.getOrDefault(PROPERTY_NAME_DESTINATION, "."));
+  }
+
+  abstract public void exportToStream(SegmentContainer sc, OutputStream stream);
+
+  abstract public boolean isExportable(SegmentContainer sc);
+
+  @Override
+  public void processSegment(SegmentContainer shot) {
+    if (!isExportable(shot)) {
+      return;
+    }
+    try {
+      /* Prepare folder and OutputStream. */
+      Path folder = this.destination.resolve(shot.getSuperId());
+      if (!folder.toFile().exists()) {
+        folder.toFile().mkdirs();
+      }
+      Path path = folder.resolve(shot.getId() + this.fileExtension);
+      OutputStream os = Files.newOutputStream(path);
+
+      /* Write audio data to OutputStream. */
+      exportToStream(shot, os);
+
+      /* Close OutputStream. */
+      os.close();
+    } catch (Exception e) {
+      LOGGER.error("Could not data to file for segment {} due to {}.", shot.getId(), LogHelper.getStackTrace(e));
+    }
+  }
+
+  public String exportToDataUrl(SegmentContainer shot) {
+    // create a ByteArrayOutputStream
+    ByteArrayOutputStream baos = new ByteArrayOutputStream();
+
+    // call the exportToStream method with the ByteArrayOutputStream
+    exportToStream(shot, baos);
+
+    // convert the ByteArrayOutputStream's data to a byte array
+    byte[] bytes = baos.toByteArray();
+
+    // encode the byte array to a Base64 string
+    String base64 = Base64.getEncoder().encodeToString(bytes);
+
+    // concatenate the data URL prefix and the Base64 string
+    return this.dataUrlPrefix + base64;
+  }
+
+  public byte[] exportToBinary(SegmentContainer shot) {
+    // create a ByteArrayOutputStream
+    ByteArrayOutputStream baos = new ByteArrayOutputStream();
+
+    // call the exportToStream method with the ByteArrayOutputStream
+    exportToStream(shot, baos);
+
+    // convert the ByteArrayOutputStream's data to a byte array
+    return baos.toByteArray();
+  }
+
+  @Override
+  public void init(PersistencyWriterSupplier phandlerSupply) { /* Nothing to init. */ }
+
+  @Override
+  public void finish() {  /* Nothing to finish. */}
```

Comment:
What about closing the stream?

Reply:
To put emphasis on this comment. I'd expect the `finish` method be `abstract` since the implementing exporter should flush and close its stream, not?